components/google_sheets/actions/add-single-row/add-single-row.mjs

```diff
@@ -32,15 +66,61 @@ export default {
         }),
       ],
     },
-    cells: {
-      propDefinition: [
-        googleSheets,
-        "cells",
-      ],
-    },
+    dataFormat,
+  },
+  async additionalProps() {
+    if (this.dataFormat == "column") {
+      if (!this.header) {
+        const rv = {};
+        rv.header = headerProp;
+        return rv;
+      } else {
+        if (this.header === "hasHeaders") {
+          const rv = {};
+          rv.header = headerProp;
+          const { values } = await this.googleSheets.getSpreadsheetValues(this.sheetId, `${this.sheetName}!1:1`);
+          for (let i = 0; i < values[0]?.length; i++) {
+            rv[`col_${i.toString().padStart(4, "0")}`] = {
+              type: "string",
+              label: values[0][i],
+              optional: true,
+            };
+          }
+          return rv;
+        } else {
+          const rv = {};
+          rv.header = headerProp;
+          rv.myColumnData = {
+            type: "string[]",
+            label: "Row Data",
+            description: "Provide a value for each cell of the row. Google Sheets accepts strings, numbers and boolean values for each cell. To set a cell to an empty value, pass an empty string.",
+          };
+          return rv;
+        }
+      }
+    } else {
+      const rv = {};
+      rv.arrayData = {
+        type: "any",
+        label: "Values",
+        description: "Pass an array that represents a row of values. Each array element will be treated as a cell (e.g., entering `[\"Foo\",1,2]` will insert a new row of data with values in 3 cells). The most common pattern is to reference an array exported by a previous step (e.g., `{{steps.foo.$return_value}}`). Google Sheets accepts strings, numbers and boolean values for each cell. To set a cell to an empty value, pass an empty string.",
+      };
+      return rv;
+    }
   },
-  async run() {
-    const cells = this.cells;
+  async run({ $ }) {
+    let cells;
+    if (this.dataFormat === "column") {
+      if (this.header === "hasHeaders") {
+        cells = Object.keys(this).filter((prop) => prop.startsWith("col_"))
+          .sort()
+          .map((prop) => this[prop]);
+      } else {
+        cells = this.myColumnData;
```

Comment:
As mentioned in this [slack thread](https://pipedream-users.slack.com/archives/C01E5KCTR16/p1641848923010900?thread_ts=1641586288.001300&cid=C01E5KCTR16), call `sanitizedArray()` on `this.myColumnData` to parse the `myColumnData` prop the same way the `arrayData` prop is?

Reply:
Looks like we have a bug here. I am checking

Reply:
No need to sanitize here :)

Reply:
When I set **Data Format** to "Enter a value for each column", **Header Row?** to "There is no header row", and **Row Data**'s structured mode to **off**, I get this error.

<img width="650" alt="noheaderrowerror" src="https://user-images.githubusercontent.com/19861096/149011969-24816645-cf66-40d9-b79b-221243031945.png">

Reply:
Ohh, I see, The non structured mode haha